### DISCUSSION TITLE
asset authorization based on file library folders

### DISCUSSION THREAD

**Discussion by geoHeil** [directus/directus#18436] [Feature Requests]
### Summary

currently, only full asset library can be public.
If we want to serve i.e. images to a website the HTML image tag does not allow to specify a bearer token. so the served images should most likely be servable for the public role.

However, right now the permissions are not fine-grained and all assets need to be set to public

### Motivation

fine grained permissions for assets when serving these for public views

### Detailed Design

add fine-grained permissions for assets based on individual file library directories

### Requirements List

fine grained permissions

### Drawbacks

none

### Alternatives

all assets served as public

### Adoption Strategy

-

**Comment by geoHeil** [comment]
The best workaround I can think of is to create a statically served site which pulls the images (with a bearer token) from directus and then is serving them from a different directory

**Comment by rijkvanzanten** [comment]
I'll close this for now, as I believe this already exists 🙂 

The permissions configuration for `Files` is the same as any other collection, so you can use any field value, including nested relational data, as the read permissions for the files collection. For example, you can set the read permissions of files to `{ "folder": { "name": { "_eq": "Public" }}` to allow all files in a folder called "Public" to be readable by the (public) role 👍🏻

**Reply by geoHeil** [reply]
Thanks
